tests/tracer/test_sampler.py

```diff
@@ -124,19 +120,17 @@ def test_sample_rate_deviation(self):
     def test_deterministic_behavior(self):
         """Test that for a given trace ID, the result is always the same"""
         tracer = DummyTracer()
-
-        tracer._sampler = RateSampler(0.5)
+        tracer._sampler = DatadogSampler(default_sample_rate=0.5, rate_limit=-1)
```

Comment:
@ZStriker19 RateSampler doesn't actually sample spans (it does not set the sampling priority it just returns True/False). So I had to use the DatadogSampler instead. Should we fix this behavior?

Reply:
We could just check to make sure the value returned by the RateSampler is the same for each span it samples with the same id.